tutorials/R/Geospatial-skills/primer-raster-r/Introduction-to-Raster-Data-In-R/Introduction-to-Raster-Data-In-R.Rmd

```diff
@@ -398,8 +406,12 @@ used to quickly define a crop extent.
 
 #plot the DEM
 plot(DEM)
+
 #Define the extent of the crop by clicking on the plot
-cropbox1 <- drawExtent()
+cropbox1 <- drawExtent(show=TRUE, col="red")
+
+#Cannot crop via directly clicking on plot
```

Comment:
Please remove this comment. Yes, that's correct that you won't be able to use the `drawExtent()` function in .Rmd documents because you can't have interactive plots (or add to plots, and many other limitations). The solution to that, however, is actually already in the code! Do you see how, at the top of this code chunk, (line 405 in this latest version) there is the code chunk argument `eval=FALSE`? That tells the .Rmd document that it shouldn't actually run this code chunk while knitting into a .md document (because it will run this section and then expect an input from the user (clicking on the plot to draw the extent) which it can't receive when 'knitting' and running through all of the code quickly. The clever second part to this solution is the next code chunk below which 'manually' defines the cropbox2 extent by plugging numbers (in this case, easting and westing) into the `extent()` function to define the extent, rather than by clicking on the map. There are advantages to both methods, but in this case, we can actually run the next code chunk (which doesn't have eval=FALSE) because it doesn't require a user input, and can run all of the code automatically without stopping during the 'knitting' process.

Reply:
@donal-at-NEON Thank you for the comments, explanations were all very helpful. I've made the changes and updated these pull requests.